src/test/event.processor.spec.ts

```diff
@@ -0,0 +1,169 @@
+
+  beforeEach(() => {
+    eventProcessor = new EventProcessor();
+    jest.clearAllMocks();
+  });
+
+  it('the instance should be defined', () => {
+    expect(eventProcessor).toBeDefined();
+  });
+
+  it('should throw an error if options are empty', async () => {
+    await expect(eventProcessor.start({}))
+      .rejects.toThrow(/No emitter address or identifier set/);
+  });
+
+  it('should throw an error if the timestamp function is undefined', async () => {
+    await expect(eventProcessor.start({
+      emitterAddresses: ['erd1nz88q5pevl6up2qxsgpqgc0qmnm93lh888wwwa68kmz363kdwz9q8tnems'],
+    })).rejects.toThrow(/Undefined function for getting the last processed timestamp/);
+  });
+
+  it('should throw an error if the timestamp function returns undefined', async () => {
+    await expect(eventProcessor.start({
+      emitterAddresses: ['erd1nz88q5pevl6up2qxsgpqgc0qmnm93lh888wwwa68kmz363kdwz9q8tnems'],
+      getLastProcessedTimestamp: async () => undefined,
+    })).rejects.toThrow(/Cannot get the last processed timestamp via the provided getLastProcessedTimestamp/);
+  });
+
+  it('should throw an error if elasticUrl is missing', async () => {
+    await expect(eventProcessor.start({
+      emitterAddresses: ['erd1nz88q5pevl6up2qxsgpqgc0qmnm93lh888wwwa68kmz363kdwz9q8tnems'],
+      getLastProcessedTimestamp: async () => 37,
+    })).rejects.toThrow(/Missing elasticUrl/);
+  });
+
+  it('should throw an error if onEventsReceived is missing', async () => {
+    await expect(eventProcessor.start({
+      emitterAddresses: ['erd1nz88q5pevl6up2qxsgpqgc0qmnm93lh888wwwa68kmz363kdwz9q8tnems'],
+      getLastProcessedTimestamp: async () => 37,
+      elasticUrl: 'https://index.multiversx.com',
+    })).rejects.toThrow(/Missing onEventsReceived callback function/);
+  });
+
+  it('should throw an error if setLastProcessedTimestamp is missing', async () => {
+    await expect(eventProcessor.start({
+      emitterAddresses: ['erd1nz88q5pevl6up2qxsgpqgc0qmnm93lh888wwwa68kmz363kdwz9q8tnems'],
+      getLastProcessedTimestamp: async () => 37,
+      elasticUrl: 'https://index.multiversx.com',
+      onEventsReceived: async () => {},
+    })).rejects.toThrow(/Missing setLastProcessedTimestamp callback function/);
+  });
+
+  it('should work but no item matches the filters', async () => {
+    const eventProcessor = new EventProcessor();
+    let receivedEvents: any[] = [];
+    const mockResponse = createElasticSearchResponse();
+
+    // Mock axios.post to return a resolved Promise with mockResponse
+    (axios.post as jest.Mock).mockResolvedValue(mockResponse);
+
+    await eventProcessor.start({
+      emitterAddresses: ['erd1nz88q5pevl6up2qxsgpqgc0qmnm93lh888wwwa68kmz363kdwz9q8tnems'],
+      getLastProcessedTimestamp: async () => 37,
+      elasticUrl: 'https://myelastic.com',
+      onEventsReceived: async (highestTimestamp, events) => {receivedEvents = events;},
+      setLastProcessedTimestamp: async (timestamp: number) => {},
+    });
+
```

Comment:
add expect for mockResponse just to check that the response from ElasticSearch is []

Reply:
No, since I won't receive anything because the es response doesn't contain any hit